src/librbd/mirror/snapshot/PromoteRequest.cc

```diff
@@ -246,13 +246,20 @@ void PromoteRequest<I>::handle_acquire_exclusive_lock(int r) {
 
 template <typename I>
 void PromoteRequest<I>::rollback() {
+  CephContext *cct = m_image_ctx->cct;
+  ldout(cct, 15) << dendl;
```

Comment:
I would still leave `ldout(cct, 15) << dendl` after `if (m_rollback_snap_id == CEPH_NOSNAP)` check.

The convention is we log the function call only when it does something (not returns earlier), i.e. when we enter the "state".